core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -362,7 +362,10 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
 
   def notYetRejoinedMembers = members.values.filter(!_.isAwaitingJoin).toList
 
-  def hasAllMembersJoined = members.size == numMembersAwaitingJoin && pendingMembers.isEmpty
+  def hasAllMembersJoined = {
+    info(s"The join complete condition checking: members => $members, members waiting join count: $numMembersAwaitingJoin, pending members: $pendingMembers")
```

Comment:
Is INFO level appropriate? (If yes, the message read a little too code centric IMHO)